### ISSUE TITLE
Perform a classifier two sample test to detect if there is covariate shift

### ISSUE THREAD

**Issue by covuworie** [covuworie/nobel-physics-prizes#71]
Perform a classifier two sample test (C2ST) to detect if there is *covariate shift* between the training and validation set (the latter acting as a proxy for test set) using both the original set of features and the ones produced from the topics. This C2ST determines if two samples are drawn from the same distribution.

Loosely follow the methodology in https://arxiv.org/pdf/1610.06545.pdf except for the following differences:
- Use MCC as test statistic.
- Use the *bootstrap* to determine confidence interval (95% or 99%) of test statistic and use this instead of the p value to reject / accept the null hypothesis that the datasets are from the same distribution.  MCC = 0 for null hypothesis since a classifier should have chance level performance at distinguishing instance from training and validation sets if they are drawn from same distribution. So if confidence interval does not contain this value reject the null.
